packages/static-hosting/lib/static-hosting.ts

```diff
@@ -1,144 +1,99 @@
-import { HostedZone, ARecord } from "@aws-cdk/aws-route53";
-import { User, Group, Policy, PolicyStatement, Effect } from "@aws-cdk/aws-iam";
-import { Version } from "@aws-cdk/aws-lambda";
-import { ArbitraryPathRemapFunction } from "./arbitrary-path-remap";
+  OriginRequestPolicy,
+  CachePolicy,
+  OriginRequestHeaderBehavior,
+  CacheHeaderBehavior,
+  IResponseHeadersPolicy,
+} from "aws-cdk-lib/aws-cloudfront";
+import { S3Origin } from "aws-cdk-lib/aws-cloudfront-origins";
+import {
+  Effect,
+  Group,
+  Policy,
+  PolicyStatement,
+  User,
+} from "aws-cdk-lib/aws-iam";
+import { ARecord, HostedZone, RecordTarget } from "aws-cdk-lib/aws-route53";
+import { CloudFrontTarget } from "aws-cdk-lib/aws-route53-targets";
+import {
+  BlockPublicAccess,
+  Bucket,
+  BucketEncryption,
+  BucketProps,
+} from "aws-cdk-lib/aws-s3";
 import { CSP } from "../types/csp";
-import { RecordTarget } from "@aws-cdk/aws-route53";
-import { CloudFrontTarget } from "@aws-cdk/aws-route53-targets";
 
 export interface StaticHostingProps {
+  domainNames: string[];
   exportPrefix?: string;
-  domainName: string;
-  subDomainName: string;
-  certificateArn: string;
-  createDnsRecord?: boolean;
-  createPublisherGroup?: boolean;
+  enforceSSL?: boolean;
+  disableCSP?: boolean;
+  enableS3AccessLogging?: boolean;
+  s3ExtendedProps?: BucketProps;
   createPublisherUser?: boolean;
-  extraDistributionCnames?: ReadonlyArray<string>;
+  createPublisherGroup?: boolean;
   enableCloudFrontAccessLogging?: boolean;
-  enableS3AccessLogging?: boolean;
-  zoneName?: string;
-  /**
-   * Used to add Custom origins and behaviors
-   */
-  customOriginConfigs?: Array<SourceConfiguration>;
-  /**
-   * Used to prepend the behaviours introduced as part of customOriginConfigs.
-   * This is a work around until this construct is updated to use the new
-   * Distribution API.
-   * https://docs.aws.amazon.com/cdk/api/latest/docs/aws-cloudfront-readme.html#distribution-api
-   */
-  prependCustomOriginBehaviours?: boolean;
-
-  /**
-   * Optional set of behaviors to override the default behavior defined in this construct
-   */
-  behaviors?: Array<Behavior>;
-  enableErrorConfig?: boolean;
-  enableStaticFileRemap?: boolean;
-  remapPaths?: remapPath[];
-  backendHost?: string;
-  remapBackendPaths?: remapPath[];
+  errorResponsePagePath?: string;
+  webAclId?: string;
   defaultRootObject?: string;
-  enforceSSL?: boolean;
-  /**
-   * Disable the use of the CSP header. Default value is false.
-   */
-  disableCSP?: boolean;
-  /**
-   * AWS limits the max header size to 1kb, this is too small for complex csp headers.
-   * The main purpose of this csp header is to provide a method of setting a report-uri.
-   */
+  certificateArn: string;
+  defaultBehaviorEdgeLambdas: EdgeLambda[];
+  additionalBehaviors?: Record<string, BehaviorOptions>;
+  enableErrorConfig?: boolean;
+  createDnsRecord?: boolean;
+  zoneName?: string;
   csp?: CSP;
-  /**
```

Comment:
Not sure why the comments have been removed?

Reply:
I've retained these :sweat_smile: